### ISSUE TITLE
Laravel 5.3 - FatalThrowableError in sitemap.php line 5: Call to a member function getLocation() on string

### ISSUE THREAD

**Issue by shamilchoudhury** [dwightwatson/sitemap#34]
Set up everything as described in the Read me section. General sitemap works, however for posts and categories, I get `FatalThrowableError in sitemap.php line 5: Call to a member function getLocation() on string` error.

Here's the controller code:

```php
class SitemapsController extends Controller
{
	public function index()
	{
		// Get a general sitemap.
		Sitemap::addSitemap('/sitemap');

		// You can use the route helpers too.
		Sitemap::addSitemap(route('sitemap.posts'));
        Sitemap::addSitemap(route('sitemap.categories'));
		// Return the sitemap to the client.
		return Sitemap::index();
	}

    public function posts()
    {
        $posts = Post::all();

        foreach ($posts as $post) {
            Sitemap::addTag(route('post.show', $post->slug), $post->updated_at, 'daily', '0.8');
        }

        return Sitemap::render();
    }

	public function categories()
	{
	    $categories = Category::all();

	    foreach ($categories as $category) {
            Sitemap::addTag(route('category.show', $category->slug), $category->updated_at, 'daily', '0.8');
        }

	    return Sitemap::render();
	}
}
```

and here are the routes : 

```php
Route::get('sitemap', 'SitemapController@index');
Route::get('/sitemap/posts', [
	'uses' => 'SitemapController@posts',
	'as' => 'sitemap.posts'
]);
Route::get('/sitemap/categories', [
	'uses' => 'SitemapController@categories',
	'as' => 'sitemap.categories'
]);
```

**Comment by dwightwatson** [comment]
I'm not able to replicate this at all - I've created a fresh Laravel 5.3 project and set up a basic scaffold just as you have. It sounds as though a tag isn't being added correctly but I can't see why that would be happening. Are you able to provide a sample project that demonstrates the issue?

**Comment by shamilchoudhury** [comment]
Hi please take a look at [this project](https://github.com/shamilchoudhury/droidtronix) here. Please clone the project (UI will be broken) and seed a few posts and categories to see the problem.

**Comment by dwightwatson** [comment]
Took a little bit of running around - but I think I found the cause. You have two view composers in your application that set the `tags` parameter in the view. Under the hood this package uses `$tags` in the sitemap view to loop through, and your view composer is overriding this. 

I've updated the library to use `$__sitemaps` and `$__tags` internally so that they are unlikely to clash with other view variables. If you would be able to test this package using `dev-master` and let me know if that fixes the issue on your end I will publish a new version.
